flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/Catalog.java

```diff
@@ -56,6 +57,16 @@
 		return Optional.empty();
 	}
 
+	/**
+	 * Get an optional {@link FunctionDefinitionFactory} instance that's responsible for generating function definitions
+	 * for catalog functions in this catalog.
+	 *
+	 * @return an optional FunctionDefinitionFactory instance
+	 */
+	default Optional<FunctionDefinitionFactory> getFunctionDefinitionFactory() {
```

Comment:
Do we need an additional method here? Both source/sink and function factories have a common root and we could just have a generic `getTableFactory()` method.

Reply:
make sense.

Side note: I wonder if we should have a better name for `Table`Factory in the near future. As far as I know, it's already used for table, catalog, and function. Though its javadoc describes itself as "A factory to create different table-related instances from string-based properties.", it's still quite confusing to developers - I myself frequently get confused by it and miss points like what you suggested, let alone those who are new and not familiar with this matter

Reply:
Agreed. I would also vote for calling it just `Factory`. But in a backwards compatible-way with proper deprecation.